packages/ember-glimmer/tests/integration/helpers/input-test.js

```diff
@@ -0,0 +1,451 @@
+    this.assertAttr('name', 'original-name');
+    this.assertAttr('maxlength', '10');
+    this.assertAttr('size', '20');
+    this.assertAttr('tabindex', '30');
+
+    this.runTask(() => {
+      set(this.context, 'value', 'Updated value');
+      set(this.context, 'disabled', true);
+      set(this.context, 'placeholder', 'Updated placeholder');
+      set(this.context, 'name', 'updated-name');
+      set(this.context, 'maxlength', 11);
+      set(this.context, 'size', 21);
+      set(this.context, 'tabindex', 31);
+    });
+
+    this.assertDisabled();
+    this.assertValue('Updated value');
+    this.assertAttr('placeholder', 'Updated placeholder');
+    this.assertAttr('name', 'updated-name');
+    this.assertAttr('maxlength', '11');
+    this.assertAttr('size', '21');
+    this.assertAttr('tabindex', '31');
+
+    this.runTask(() => {
+      set(this.context, 'value', 'Original value');
+      set(this.context, 'disabled', false);
+      set(this.context, 'placeholder', 'Original placeholder');
+      set(this.context, 'name', 'original-name');
+      set(this.context, 'maxlength', 10);
+      set(this.context, 'size', 20);
+      set(this.context, 'tabindex', 30);
+    });
+
+    this.assertNotDisabled();
+    this.assertValue('Original value');
+    this.assertAttr('placeholder', 'Original placeholder');
+    this.assertAttr('name', 'original-name');
+    this.assertAttr('maxlength', '10');
+    this.assertAttr('size', '20');
+    this.assertAttr('tabindex', '30');
+  }
+
+  ['@test static attributes']() {
+    this.render(`
+      {{input type="text"
+        disabled=true
+        value="Original value"
+        placeholder="Original placeholder"
+        name="original-name"
+        maxlength=10
+        size=20
+        tabindex=30
+      }}`
+    );
+
+    this.assertDisabled();
+    this.assertValue('Original value');
+    this.assertAttr('placeholder', 'Original placeholder');
+    this.assertAttr('name', 'original-name');
+    this.assertAttr('maxlength', '10');
+    this.assertAttr('size', '20');
+    this.assertAttr('tabindex', '30');
+
+    this.runTask(() => this.rerender());
+
+    this.assertDisabled();
+    this.assertValue('Original value');
+    this.assertAttr('placeholder', 'Original placeholder');
+    this.assertAttr('name', 'original-name');
+    this.assertAttr('maxlength', '10');
+    this.assertAttr('size', '20');
+    this.assertAttr('tabindex', '30');
+  }
+
+  ['@test cursor selection range'](assert) {
+    this.render(`{{input type="text" value=value}}`, { value: 'original' });
+
+    let input = this.$input()[0];
+
+    //TODO: GJ: this is (0, 0) on Firefox:
+    // See https://ember-twiddle.com/33e506329f8176ae874422644d4cc08c?openFiles=components.input-component.js%2Ctemplates.components.input-component.hbs
+    // this.assertSelectionRange(8, 8);
+
+    this.runTask(() => this.rerender());
+
+    //TODO: GJ: this is (0, 0) on Firefox
+    // this.assertSelectionRange(8, 8);
```

Comment:
this looks like it might be a bug too: [twiddle](https://ember-twiddle.com/33e506329f8176ae874422644d4cc08c?openFiles=components.input-component.js%2Ctemplates.components.input-component.hbs)